### PR TITLE
Fix introspection endpoint url in tenant legacy URL mode in OIDC config response

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request
- Resolves: https://github.com/wso2/product-is/issues/8521
- This will add tenant as a path param to introspection endpoint in OIDC discovery response in legacy tenant URL mode.

### When should this PR be merged
This PR should be merged before https://github.com/wso2/product-is/pull/8527

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 1]
```diff
@@ -1264,6 +1264,26 @@ public static String getOAuth2IntrospectionEPUrl() {
...
+         * @return String of the resolved URL for the introspection endpoint.
+         * @throws URISyntaxException URI Syntax Exception.
+         */
+        public static String getOAuth2IntrospectionEPUrl(String tenantDomain) throws URISyntaxException {
+
+            String getOAuth2IntrospectionEPUrl = buildUrl(OAUTH2_INTROSPECT_EP_URL,
+                    OAuthServerConfiguration.getInstance()::getOauth2IntrospectionEPUrl);
+
+            if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && isNotSuperTenant(tenantDomain)) {
+                //Append tenant domain to path when the tenant-qualified url mode is disabled.
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:1280]: add space after //

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1399/commits/b67eb489e485f9f7fc6f19c1526ceaf8fea2fa45

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 2]
```diff
@@ -1264,6 +1264,26 @@ public static String getOAuth2IntrospectionEPUrl() {
...
+         * @throws URISyntaxException URI Syntax Exception.
+         */
+        public static String getOAuth2IntrospectionEPUrl(String tenantDomain) throws URISyntaxException {
+
+            String getOAuth2IntrospectionEPUrl = buildUrl(OAUTH2_INTROSPECT_EP_URL,
+                    OAuthServerConfiguration.getInstance()::getOauth2IntrospectionEPUrl);
+
+            if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && isNotSuperTenant(tenantDomain)) {
+                // Append tenant domain to path when the tenant-qualified url mode is disabled.
+                getOAuth2IntrospectionEPUrl =
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:1281]: Why do we need the tenantDomain as a query param in the introspection endpoint in legacy mode?

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:1281]: With this PR, the tenant domain added as a path param to the introspection endpoint in the legacy tenant URL mode.

**Pull request merged**